src/mpid/ch4/generic/am/mpidig_am_send.h

```diff
@@ -104,20 +101,10 @@ static inline int MPIDIG_do_ssend(const void *buf, MPI_Aint count, MPI_Datatype
     MPIDIG_REQUEST(sreq, count) = count;
 #endif
 
-    /* Increment the completion counter once to account for the extra message that needs to come
-     * back from the receiver to indicate completion. */
-    MPIR_cc_incr(sreq->cc_ptr, &c);
+    mpi_errno = MPIDIG_isend_impl_new(buf, count, datatype, rank, tag, comm, context_offset, addr,
```

Comment:
I think this is a lot of churn because it is doing 2 things at once. SSEND now uses RNDV in all cases. We could change that to do eager/rndv independently of this PR. Switching to the new protocol framework would then be somewhat easier to digest.

Reply:
Actually, the SSEND today is ignoring RNDV path and uses the NM/SHM isend directly. It is relying on that fact that NM/SHM can handle long message internally. Technically, SSEND cannot use the existing RNDV protocol at all because the `MPIDIG_SSEND_REQ` handler id would not be sent over the RNDV protocol (it will send `MPIDIG_SEND_LONG_REQ` instead).

I guess I could create an `MPIDIG_do_ssend_new` for this PR, so we doing new function rather than changing an existing one.